Build failure: nvidiaPackages.legacy_470

with nixpkgs 3c5319ad3aa51551182ac82ea17ab1c6b0f0df89

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `nvidiaPackages.legacy_470`

### Build log

```
/build/kernel/nvidia/nv-acpi.c:84:19: error: initialization of 'void (*)(struct acpi_device *)' from incompatible pointer type 'int (*)(struct acpi_device *, int)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
   84 |         .remove = nv_acpi_remove_two_args,
      |                   ^~~~~~~~~~~~~~~~~~~~~~~
/build/kernel/nvidia/nv-acpi.c:84:19: note: (near initialization for 'nv_acpi_driver_template.ops.remove')
  CC [M]  /build/kernel/nvidia/nv-i2c.o
cc1: some warnings being treated as errors
make[3]: *** [/nix/store/x6vcbd4y6h3hyp0npzrgpak3f5n3kvni-linux-6.2.2-dev/lib/modules/6.2.2/source/scripts/Makefile.build:252: /build/kernel/nvidia/nv-acpi.o] Error 1
make[3]: *** Waiting for unfinished jobs....
/build/kernel/nvidia/nv-dma.c:963: warning: "IMPORT_SGT_STUBS_NEEDED" redefined
  963 | #define IMPORT_SGT_STUBS_NEEDED 0
      |
/build/kernel/nvidia/nv-dma.c:957: note: this is the location of the previous definition
  957 | #define IMPORT_SGT_STUBS_NEEDED 1
      |
make[2]: *** [/nix/store/x6vcbd4y6h3hyp0npzrgpak3f5n3kvni-linux-6.2.2-dev/lib/modules/6.2.2/source/Makefile:2021: /build/kernel] Error 2
make[2]: Leaving directory '/nix/store/x6vcbd4y6h3hyp0npzrgpak3f5n3kvni-linux-6.2.2-dev/lib/modules/6.2.2/build'
make[1]: *** [Makefile:242: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/x6vcbd4y6h3hyp0npzrgpak3f5n3kvni-linux-6.2.2-dev/lib/modules/6.2.2/source'
make: *** [Makefile:80: modules] Error 2
error: builder for '/nix/store/28gj3lyzcyv7qh9y31f4q8m1pp2wj1x9-nvidia-x11-470.161.03-6.2.2.drv' failed with exit code 2;
       last 10 log lines:
       >   963 | #define IMPORT_SGT_STUBS_NEEDED 0
       >       |
       > /build/kernel/nvidia/nv-dma.c:957: note: this is the location of the previous definition
       >   957 | #define IMPORT_SGT_STUBS_NEEDED 1
       >       |
       > make[2]: *** [/nix/store/x6vcbd4y6h3hyp0npzrgpak3f5n3kvni-linux-6.2.2-dev/lib/modules/6.2.2/source/Makefile:2021: /build/kernel] Error 2
       > make[2]: Leaving directory '/nix/store/x6vcbd4y6h3hyp0npzrgpak3f5n3kvni-linux-6.2.2-dev/lib/modules/6.2.2/build'
       > make[1]: *** [Makefile:242: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/x6vcbd4y6h3hyp0npzrgpak3f5n3kvni-linux-6.2.2-dev/lib/modules/6.2.2/source'
       > make: *** [Makefile:80: modules] Error 2
       For full logs, run 'nix log /nix/store/28gj3lyzcyv7qh9y31f4q8m1pp2wj1x9-nvidia-x11-470.161.03-6.2.2.drv'.
error: 1 dependencies of derivation '/nix/store/ykhm772q783l873y8xsz8v8brj62ad3r-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/i4jsjqbrywqixrrqw1b0dkxcz5x1lzvl-linux-6.2.2-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/chbdam47hg03lh62ll4ll3zhylp5qn9m-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ir38k80mzhgrp1i815zxcn3gg0c455da-nixos-system-desktop-23.05.20230304.3c5319a.drv' failed to build
```

### Additional context

kernel 6.1.10

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@PedroHLC @Kiskae

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.10, NixOS, 23.05 (Stoat), 23.05.20230207.fab0908`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.2`
 - nixpkgs: `/run/current-system/nixpkgs`

--- a/pkgs/os-specific/linux/nvidia-x11/default.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/default.nix
@@ -71,6 +71,22 @@ rec {
     sha256_64bit = "sha256-Xagqf4x254Hn1/C+e3mNtNNE8mvU+s+avPPHHHH+dkA=";
     settingsSha256 = "sha256-ryUSiI8PsY3knkJLg0k1EmyYW5OWkhuZma/hmXNuojw=";
     persistencedSha256 = "sha256-/2h90Gq9NQd9Q+9eLVE6vrxXmINXxlLcSNOHxKToOEE=";
+
+    prePatch = "pushd kernel";
+    postPatch = "popd";
+
+    patches = [
+      # source: https://gist.github.com/joanbm/963906fc6772d8955faf1b9cc46c6b04
+      (fetchpatch {
+        url = "https://gist.github.com/joanbm/963906fc6772d8955faf1b9cc46c6b04/raw/0f99aa10d47b524aa0e6e3845664deac3a1ad9d9/nvidia-470xx-fix-linux-6.2.patch";
+        hash = "sha256-5n5/4ivK8od8EJNJf0PI9ZZ4U5RjOw+h4HakA+lmW1c=";
+      })
+      # source: https://gist.github.com/joanbm/d10e9cbbbb8e245b6e7e27b2db338faf
+      (fetchpatch {
+        url = "https://gist.github.com/joanbm/d10e9cbbbb8e245b6e7e27b2db338faf/raw/f5d5238bdbaa16cd4008658a0f82b9dd84f1b38f/nvidia-470xx-fix-linux-6.3.patch";
+        hash = "sha256-mR+vXDHgVhWC0JeLgGlbNVCH8XTs7XnhEJS6BV75tI8=";
+      })
+    ];
   };
 
   # Last one supporting x86

--- a/pkgs/os-specific/linux/nvidia-x11/generic.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/generic.nix
@@ -13,6 +13,7 @@
 , settings32Bit ? false
 
 , prePatch ? ""
+, postPatch ? null
 , patches ? []
 , broken ? false
 , brokenOpen ? broken
@@ -68,7 +69,7 @@ let
       else throw "nvidia-x11 does not support platform ${stdenv.hostPlatform.system}";
 
     patches = if libsOnly then null else patches;
-    inherit prePatch;
+    inherit prePatch postPatch;
     inherit version useGLVND useProfiles;
     inherit (stdenv.hostPlatform) system;
     inherit i686bundled;

